### PR TITLE
chore: catch error when demo server start

### PR DESCRIPTION
**Pull request by Frapschen**
It's better to catch any error during the demo OpAMP Server starting. e.g. the 4320 port already in use.

before this PR: if we start an OpAMP Server in the background, and then start another OpAMP Server, we will get:
```
fraps@CMQ server % ./bin/server 
2024/01/10 10:46:33.961345 [MAIN] OpAMP Server starting...
2024/01/10 10:46:33.963579 [MAIN] OpAMP Server running...
```
however, it fails to start the OpAMP Server.

with this PR's enhance:
```
fraps@CMQ server % ./bin/server
2024/01/10 10:49:12.296051 [MAIN] OpAMP Server starting...
2024/01/10 10:49:12.298014 [OPAMP] OpAMP server start fail: listen tcp 127.0.0.1:4320: bind: address already in use
```

It response the real status of the OpAMP server.

### CODE REVIEW

**Review by srikanthccv — APPROVED** [review]
Thank you

File: client/types/logger.go [hunk 1]
```diff
@@ -4,4 +4,5 @@ package types
 type Logger interface {
 	Debugf(format string, v ...interface{})
 	Errorf(format string, v ...interface{})
+	Fatalf(format string, v ...interface{})
```
Comment (tigrannajaryan) [inline client/types/logger.go:7]: I would not want to change the Logger interface which is part of public API. This is a breaking change and also increases the public API size.

Instead I suggest where necessary the example code calls the regular Logger.Errorf() followed by a whatever its preferred termination function is (e.g. os.Exit).

File: internal/examples/server/opampsrv/opampsrv.go [hunk 2]
```diff
@@ -64,7 +64,9 @@ func (srv *Server) Start() {
 	}
 	settings.TLSConfig = tlsConfig
 
-	srv.opampSrv.Start(settings)
+	if err := srv.opampSrv.Start(settings); err != nil {
+		srv.logger.Fatalf("OpAMP server start fail: %v", err.Error())
```
Comment (tigrannajaryan) [inline internal/examples/server/opampsrv/opampsrv.go:68]:
```suggestion
		srv.logger.Errorf("OpAMP server start fail: %v", err.Error())
		os.Exit(1)
```

**Review by tigrannajaryan — CHANGES_REQUESTED** [review]

File: client/types/logger.go (hunk 1 above)
Comment (Frapschen) [reply client/types/logger.go]: ok

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Thank you @Frapschen

**Pull request merged**